src/psygnal/_signal.py

```diff
@@ -1034,14 +1040,25 @@ def __call__(
     def _run_emit_loop(self, args: tuple[Any, ...]) -> None:
         # allow receiver to query sender with Signal.current_emitter()
         with self._lock:
-            with Signal._emitting(self):
-                for caller in self._slots:
-                    try:
-                        caller.cb(args)
-                    except Exception as e:
-                        raise EmitLoopError(
-                            cb=caller, args=args, exc=e, signal=self
-                        ) from e
+            self._emit_queue.append(args)
```

Comment:
my first thought here is that the new `_emit_queue` seems awfully similar to the existing `_args_queue`, which already deals with the concept of queuing emission events.  So, perhaps we can unify those two mechanisms?  

I think that would mean that this line `self._emit_queue.append(args)` would move up to the `emit` method (i.e. it would pull the existing `self._args_queue.append(args)` call on line ~981 out of the the `if self._is_paused:` conditional)... and then `_run_emit_loop` would look more like `while self._args_queue: args = self._args_queue.pop()...`

Reply:
and then, it also wouldn't need to be an exception to emit during emission... it would just assume the proper place at the end of the queue

Reply:
The exception is to allow preventing of object modification during signal emission. It may be useful for debug, but also for some fragile objects.  

In context of merge queues 

Lets see this code:

```python

s = EventedSet()

def callback1():
   if 1 not in s:
       s.add(1)

def callback2():
   if 5 not in s:
       s.update(range(5, 10)

s.events.items_changed.connect(callback1) 
s.events.items_changed.connect(callback2)

s.add(11)
```

What events you expect to be emitted

```
((11,),()),  ((1,),()), ((5,6,7,8,9),())
```
or 

```
((11,),()), ((1,5,6,7,8,9),())
```

As after merge queues, I do not see option to generate first one that is more accurate for me.

Reply:
I think we're in agreement that the first emission pattern is the desired pattern (with (1,) separate from the 5-10). But I'm unsure what you mean by "merge queues".  I don't think that calling emit should merge the current emitted args with any existing ones, but rather that it should be added to a queue of emission.  Lemme see if I can put into code what I'm imagining.